src/search/landmarks/landmark_factory_relaxation.cc

```diff
@@ -9,32 +11,20 @@ void LandmarkFactoryRelaxation::generate_landmarks(const shared_ptr<AbstractTask
     TaskProxy task_proxy(*task);
     Exploration exploration(task_proxy);
     generate_relaxed_landmarks(task, exploration);
-    generate(task_proxy, exploration);
+    postprocess(task_proxy, exploration);
 }
 
-void LandmarkFactoryRelaxation::generate(const TaskProxy &task_proxy, Exploration &exploration) {
-    if (only_causal_landmarks)
-        discard_noncausal_landmarks(task_proxy, exploration);
-    if (!disjunctive_landmarks)
-        discard_disjunctive_landmarks();
-    if (!conjunctive_landmarks)
-        discard_conjunctive_landmarks();
+void LandmarkFactoryRelaxation::postprocess(const TaskProxy &task_proxy, Exploration &exploration) {
     lm_graph->set_landmark_ids();
-
-    if (no_orders)
-        discard_all_orderings();
-    else if (reasonable_orders) {
-        utils::g_log << "approx. reasonable orders" << endl;
-        approximate_reasonable_orders(task_proxy, false);
-        utils::g_log << "approx. obedient reasonable orders" << endl;
-        approximate_reasonable_orders(task_proxy, true);
-    }
     mk_acyclic_graph();
     calc_achievers(task_proxy, exploration);
 }
 
 void LandmarkFactoryRelaxation::discard_noncausal_landmarks(
     const TaskProxy &task_proxy, Exploration &exploration) {
```

Comment:
I'm not sure if axioms mess up the causal landmarks computation.
I'm also not sure if we should include a guard for conjunctive landmarks. We know that they are not handled properly since lm_hm(m=2,only_causal_landmarks=true) pruned landmarks while it should only produce causal landmarks in the first place. Not having a guard is currently not a problem since we removed only_causal_landmarks as an option from the only factory that created conjunctive landmarks; but if someone comes up with a new way of computing conjunctive landmarks and wants to use this option there's no indication that it won't work correctly.

Reply:
Without this comment from the pull request, I wouldn't understand the TODO comment.

Reply:
I suggest adding some explanation to the TODO comment in the code.